pkg/upgrade/control_plane.go

```diff
@@ -175,6 +175,20 @@ func (u *ControlPlaneUpgrader) Upgrade() error {
 		return errors.New("Found 0 control plane machines")
 	}
 
+	// Begin the upgrade by reconciling the kubeadm ConfigMap's ClusterStatus.APIEndpoints, just in case the data
+	// is out of sync.
+	u.log.Info("Listing workload cluster Nodes")
+	nodeList, err := u.targetKubernetesClient.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error listing workload cluster nodes")
+	}
+	u.log.Info("Reconciling kubeadm ConfigMap's ClusterStatus.APIEndpoints")
+	if err := u.updateKubeadmConfigMap(func(in *v1.ConfigMap) (*v1.ConfigMap, error) {
+		return reconcileKubeadmConfigMapClusterStatusAPIEndpoints(in, nodeList, machines)
+	}); err != nil {
+		return errors.Wrap(err, "error reconciling kubeadm ConfigMap")
+	}
```

Comment:
If we move all of this logic into its own method, we could just call it twice instead of replicating it in two places

Reply:
Good idea, will do